src/playlist-maker.js

```diff
@@ -248,11 +248,23 @@ const factory = (player, initialList, initialIndex = 0) => {
      */
     next() {
 
-      // Make sure we don't go past the end of the playlist.
-      let index = Math.min(playlist.currentIndex_ + 1, list.length - 1);
+      let nextIndex;
 
-      if (index !== playlist.currentIndex_) {
-        return list[playlist.currentItem(index)];
+      // Repeat
+      if (playlist.repeat_ === true) {
```

Comment:
My thinking was that I want it to not work if someone manually sets something besides `true` for repeat.